modmail/extensions/extension_manager.py

```diff
@@ -0,0 +1,276 @@
+from modmail.utils.extensions import EXTENSIONS, unqualify
+
+log: ModmailLogger = logging.getLogger(__name__)
+
+
+BASE_PATH_LEN = __name__.count(".")
+
+EXT_METADATA = ModeMetadata(production=True, develop=True, plugin_dev=True)
+
+
+class Action(Enum):
+    """Represents an action to perform on an extension."""
+
+    # Need to be partial otherwise they are considered to be function definitions.
+    LOAD = functools.partial(ModmailBot.load_extension)
+    UNLOAD = functools.partial(ModmailBot.unload_extension)
+    RELOAD = functools.partial(ModmailBot.reload_extension)
+
+
+class ExtensionConverter(commands.Converter):
+    """
+    Fully qualify the name of an extension and ensure it exists.
+
+    The * and ** values bypass this when used with the reload command.
+    """
+
+    source_list = EXTENSIONS
+    type = "extension"
+
+    async def convert(self, ctx: Context, argument: str) -> str:
+        """Fully qualify the name of an extension and ensure it exists."""
+        # Special values to reload all extensions
+        if argument == "*" or argument == "**":
+            return argument
+
+        argument = argument.lower()
+
+        if argument in self.source_list:
+            return argument
+
+        qualified_arg = f"modmail.{self.type}s.{argument}"
+        if qualified_arg in self.source_list:
+            return qualified_arg
+
+        matches = []
+        for ext in self.source_list:
+            if argument == unqualify(ext):
+                matches.append(ext)
+
+        if not matches:
+            raise commands.BadArgument(f":x: Could not find the {self.type} `{argument}`.")
+
+        if len(matches) > 1:
+            names = "\n".join(sorted(matches))
+            raise commands.BadArgument(
+                f":x: `{argument}` is an ambiguous {self.type} name. "
+                f"Please use one of the following fully-qualified names.```\n{names}```"
+            )
+
+        return matches[0]
+
+
+class ExtensionManager(ModmailCog):
+    """Extension management base class."""
+
+    type = "extension"
+
+    def __init__(self, bot: ModmailBot):
+        self.bot = bot
+        self.all_extensions = EXTENSIONS
+
+    async def get_black_listed_extensions() -> list:
+        """Returns a list of all blacklisted extensions."""
+        raise NotImplementedError()
+
+    @commands.group("ext", aliases=("extensions", "exts"), invoke_without_command=True)
+    async def extensions_group(self, ctx: Context) -> None:
+        """Load, unload, reload, and list loaded extensions."""
+        await ctx.send_help(ctx.command)
+
+    @extensions_group.command(name="load", aliases=("l",))
+    async def load_extensions(self, ctx: Context, *extensions: ExtensionConverter) -> None:
+        """
+        Load extensions given their fully qualified or unqualified names.
+
+        If '\*' or '\*\*' is given as the name, all unloaded extensions will be loaded.
+        """  # noqa: W605
```

Comment:
Not needed.

```suggestion
        """
        Load extensions are given their fully qualified or unqualified names.

        If '*' or '**' is given as the name, all unloaded extensions will be loaded.
        """
```

Reply:
It's needed due to markdown,  as the docstring is used as the help info